designer/server/src/main/scala/pl/touk/nussknacker/ui/process/repository/activities/DbScenarioActivityRepository.scala

```diff
@@ -875,17 +876,45 @@ class DbScenarioActivityRepository(override protected val dbRef: DbRef, clock: C
         )).map((entity.id, _))
 
       case ScenarioActivityType.CustomAction(actionName) =>
-        ScenarioActivity
+        (for {
+          comment <- commentFromEntity(entity)
+        } yield ScenarioActivity
           .CustomAction(
             scenarioId = scenarioIdFromEntity(entity),
             scenarioActivityId = entity.activityId,
             user = userFromEntity(entity),
             date = entity.createdAt.toInstant,
             scenarioVersion = entity.scenarioVersion,
             actionName = actionName,
-          )
-          .asRight
-          .map((entity.id, _))
+            comment = comment,
+          )).map((entity.id, _))
+    }
+  }
+
+  // todo NU-1772: in next phase the legacy comments will be fully migrated to scenario activities,
+  //  until next PR is merged there is parsing from comment content to preserve full compatibility
+  private def extractOldNameAndNewNameForRename(
+      entity: ScenarioActivityEntityData
+  ): Either[String, (String, String)] = {
+    val fromAdditionalProperties = for {
+      oldName <- additionalPropertyFromEntity(entity, "oldName")
+      newName <- additionalPropertyFromEntity(entity, "newName")
+    } yield (oldName, newName)
+
+    val legacyCommentPattern = """Rename: \[(.+?)\] -> \[(.+?)\]""".r
```

Comment:
I wonder whether NU was adding comments in such format or any external system? If it was any external system then this logic should not be here and maybe performed manually in target system.

Reply:
- Before this PR, NU was adding such comments itself, in `pl.touk.nussknacker.ui.process.repository.ProcessRepository`.
- In this PR it is replaced with adding `ScenarioActivity.ScenarioNameChanged` instead of plain text comment
- But old comments are not fully migrated in the db - because it would require parsing comment during the migration and extracting oldName and newName


Out of 17 types of activities, that are supposed to be handled according to analysis od NU-1772:
- 10 are fully migrated and handled in this PR
- 4 were not saved at all (ScenarioCreated, ChangedProcessingMode, OutgoingMigration, PerformedScheduledExecution), will be added in next PR
- 2 were saved, and are still saved, as plain text comments, not named activities (IncomingMigration, AutomaticUpdate)
- 1 remaining, the ScenarioNameChanged, was added in this PR (but old comments will have to be migrated alongise IncomingMigration and AutomaticUpdate comments in next PR, then this todo will be resolved)

In the next PR, ETA next week, I will add those 4 activities that were not saved until now, and will migrate those 3, that were plain text comments until now.